diff --git a/bin/kmstool-instance/main.c b/bin/kmstool-instance/main.c
index af52fdd..dce8ac1 100644
--- a/bin/kmstool-instance/main.c
+++ b/bin/kmstool-instance/main.c
@@ -6,0 +7 @@
+#include <aws/common/hash_table.h>
diff --git a/source/rest.c b/source/rest.c
index 5086868..1c6bb57 100644
--- a/source/rest.c
+++ b/source/rest.c
@@ -412,0 +413 @@ static void s_on_sign_complete(struct aws_signing_result *signing_result, int er
+    struct aws_http_stream *stream = NULL;
@@ -433 +434 @@ static void s_on_sign_complete(struct aws_signing_result *signing_result, int er
-    struct aws_http_stream *stream = aws_http_connection_make_request(ctx->rest_client->connection, &request_options);
+    stream = aws_http_connection_make_request(ctx->rest_client->connection, &request_options);
@@ -456,0 +458 @@ struct aws_nitro_enclaves_rest_response *aws_nitro_enclaves_rest_client_request_
+    struct aws_signable *sign_request = NULL;
@@ -479 +481 @@ struct aws_nitro_enclaves_rest_response *aws_nitro_enclaves_rest_client_request_
-    struct aws_signable *sign_request = aws_signable_new_http_request(rest_client->allocator, ctx.request);
+    sign_request = aws_signable_new_http_request(rest_client->allocator, ctx.request);
